CMakeLists.txt

```diff
@@ -48,6 +48,7 @@ if(WIN32)
     -D_WIN32_WINNT=${WIN32_WINNT}
     -DBOOST_THREAD_PROVIDES_GENERIC_SHARED_MUTEX_ON_WIN
     -DBOOST_THREAD_V2_SHARED_MUTEX
```

Comment:
Do we still need to use shared mutex from Boost with mingw-llvm?

Reply:
Not necessarily, I think we can add a check on CMAKE_CXX_COMPILER_ID.